File: Week06/d32/classes/rectangle.js
// var area = function(height, width) {
//   return (height * width)
// }

// var rectangle1 = {height: 5, width: 3};
// var rectangle2 = {height: 8, width: 9};

// console.log(area(rectangle1.height, rectangle1.width));
// console.log(area(rectangle2.height, rectangle2.width));

var Rectangle = function(height, width) {
  this.height = height;
  this.width = width;
}

Rectangle.prototype.area = function() {
  return(this.height * this.width);
}


var r1 = new Rectangle(3,6);
var r2 = new Rectangle(6,12);

console.log(r1);
console.log(r1.area());
console.log('***********');
console.log(r2);
console.log(r2.area());

File: Week06/d32/classes/rectangle.rb
class Rectangle
  attr_accessor :width, :height

  def initialize (width, height)
    @width = width
    @height = height
  end

  def area
    @width * @height
  end
end

r1 = Rectangle.new(10,12)
r2 = Rectangle.new(5,9)

puts r1
puts r1.inspect
r1.height = 15
p r1
puts r1.area
puts '*'*12
puts r2
puts r2.inspect
p r2
puts r2.area

